lib/src/cpp/utils/XMLManager.cpp

```diff
@@ -588,6 +594,9 @@ void XMLManager::move_to_node(
         throw ElementNotFound(exception_message);
     }
 
+    // TODO add a error check if any of the 'tag_name', 'name' and 'value' parameters are empty.
+    //      If happens, throw a BadParameter exception
```

Comment:
I do not think this is correct. It is an internal method so there is no need to have safety checks because the developers are in charge of keeping the contract (it is not public API so accessible to misusage by external users). What I was trying to say is that the provided `name` is empty, then the current node has to be kept. Similar to the case in a list collection where the `index` is empty. For instance `fastddsqosprof file.xml clear participant[profile].properties[]` should clear every Participant properties. Whereas if `fastddsqosprof file.xml clear participant[profile].properties[name]` is provided, then only the specific property is removed. The `DomainParticipant::clear_property` function called is exactly the same but providing an empty `key` or not.

Reply:
Ok, now I understand your point.
My idea was to check if `name` was empty before calling this method, like this:
```if (!name.empty())
{
    manager.move_to_node(tag::PROPERTY, name);
}
manager.clear_node();```
But checking empry `name` in this method and also in the other one without attributes, would simply so much the API methods (would avoid all those `if` statements in each method, so we have them only once, in the infraestructure).